include/cpr/ssl_options.h

```diff
@@ -14,6 +20,324 @@ class VerifySsl {
+  public:
+    Alpn(bool p_enabled = true) : enabled(p_enabled) {}
+
+    bool enabled;
+};
+
+//  This option enables/disables NPN in the SSL handshake (if the SSL backend libcurl is built to
+//  use supports it), which can be used to negotiate http2.
+class Npn {
+  public:
+    Npn(bool p_enabled = true) : enabled(p_enabled) {}
+
+    bool enabled;
+};
+
+// This option determines whether libcurl verifies that the server cert is for the server it is
+// known as.
+class VerifyHost {
+  public:
+    VerifyHost(bool p_enabled = true) : enabled(p_enabled) {}
+
+    bool enabled;
+};
+
+// This option determines whether libcurl verifies the authenticity of the peer's certificate.
+class VerifyPeer {
+  public:
+    VerifyPeer(bool p_enabled = true) : enabled(p_enabled) {}
+
+    bool enabled;
+};
+
+// This option determines whether libcurl verifies the status of the server cert using the
+// "Certificate Status Request" TLS extension (aka. OCSP stapling).
+class VerifyStatus {
+  public:
+    VerifyStatus(bool p_enabled = true) : enabled(p_enabled) {}
+
+    bool enabled;
+};
+
+struct TLSv1 {};
+#if LIBCURL_VERSION_MAJOR >= 7
+#if LIBCURL_VERSION_MINOR < 18
+struct SSLv2 {};
+#endif
+#if LIBCURL_VERSION_MINOR < 39
+struct SSLv3 {};
+#endif
+#if LIBCURL_VERSION_MINOR >= 34
+struct TLSv1_0 {};
+struct TLSv1_1 {};
+struct TLSv1_2 {};
+#endif // LIBCURL_VERSION_MINOR >= 34
+#if LIBCURL_VERSION_MINOR >= 52
+struct TLSv1_3 {};
+#endif
+#if LIBCURL_VERSION_MINOR >= 54
+// The flag defines maximum supported TLS version as TLSv1.0. (Added in 7.54.0)
+struct MaxTLSv1_0 {};
+// The flag defines maximum supported TLS version as TLSv1.1. (Added in 7.54.0)
+struct MaxTLSv1_1 {};
+// The flag defines maximum supported TLS version as TLSv1.2. (Added in 7.54.0)
+struct MaxTLSv1_2 {};
+// The flag defines maximum supported TLS version as TLSv1.3. (Added in 7.54.0)
+struct MaxTLSv1_3 {};
+#endif
+#endif // LIBCURL_VERSION_MAJOR >= 7
+
+// path to Certificate Authority (CA) bundle
+class CaInfo {
+  public:
+    template <typename FileType>
+    CaInfo(FileType&& p_filename) : filename(CPR_FWD(p_filename)) {}
+
+    std::string filename;
+};
+
+// specify directory holding CA certificates
+class CaPath {
+  public:
+    template <typename FileType>
+    CaPath(FileType&& p_filename) : filename(CPR_FWD(p_filename)) {}
+
+    std::string filename;
+};
+
+// specify a Certificate Revocation List file
+class Crl {
+  public:
+    template <typename FileType>
+    Crl(FileType&& p_filename) : filename(CPR_FWD(p_filename)) {}
+
+    std::string filename;
+};
+
+// specify ciphers to use for TLS
+class Ciphers {
+  public:
+    template <typename T>
+    Ciphers(T&& p_ciphers) : ciphers(CPR_FWD(p_ciphers)) {}
+
+    std::string ciphers;
+};
+
+// enable/disable use of the SSL session-ID cache
+class SessionIdCache {
+  public:
+    SessionIdCache(bool p_enabled = true) : enabled(p_enabled) {}
+
+    bool enabled;
+};
+
+} // namespace ssl
+
+struct SslOptions {
```

Comment:
Would it be possible to add an auto discover method for the system cert chain?
Are you aware of anything that can archive this?

Reply:
If we ignore the [CURLOPT_CAINFO](https://curl.haxx.se/libcurl/c/CURLOPT_CAINFO.html), libcurl will always use the system cret chain.

> (iOS and macOS) When curl uses Secure Transport this option is supported. If the option is not set, then curl will use the certificates in the system and user Keychain to verify the peer.
>
> (Schannel) This option is supported for Schannel in Windows 7 or later but we recommend not using it until Windows 8 since it works better starting then. If the option is not set, then curl will use the certificates in the Windows' store of root certificates (the default for Schannel).

> ## Default
> Built-in system specific. When curl is built with Secure Transport or Schannel, this option is not set by default.

Reply:
Perfect!